apps/frontend/src/hooks/useCacheCall.ts

```diff
@@ -0,0 +1,70 @@
+import { DependencyList, useEffect, useMemo, useState } from 'react';
+
+import { Subscription } from 'rxjs';
+
+import { defaultChainId } from '../config/chains';
+
+import {
+  CacheCallOptions,
+  observeCall,
+  startCall,
+} from '../store/rxjs/provider-cache';
+import { useBlockNumber } from './useBlockNumber';
+import { useIsMounted } from './useIsMounted';
+
+type State<T> = {
+  value: T;
+  loading: boolean;
+  error: Error | null;
+};
+
+export const useCacheCall = <T>(
+  key: string,
+  callback: () => Promise<T>,
+  deps?: DependencyList,
+  defaultValue?: T,
+  options?: Partial<CacheCallOptions>,
+): State<T> => {
+  const isMounted = useIsMounted();
+  const { value: block } = useBlockNumber(defaultChainId);
+
+  const [state, setState] = useState<State<T>>({
+    value: defaultValue ?? (null as T),
+    loading: false,
+    error: null,
+  });
+
+  useEffect(() => {
+    if (!isMounted() || !block) {
+      return;
+    }
+
+    let sub: Subscription;
+
+    sub = observeCall(`call:${key}`).subscribe(e => {
+      if (isMounted()) {
+        setState({
+          ...e.result,
+          value: e.result.value ?? defaultValue ?? (null as T),
+        });
+      }
+    });
+
+    try {
+      startCall(`call:${key}`, callback, {
+        ...options,
+        blockNumber: options?.blockNumber || block,
+      });
+    } catch (e) {}
```

Comment:
I added this try/catch block to avoid `promise(...).then() undefined` errors when no wallet is connected on `/earn/lend` page, double check if it's ideal implementation and adjust if you need